client/clientauth.js

```diff
@@ -58,11 +57,14 @@ module.exports = ( fastify, opts, done ) =>
 						host: "https://r2-pc.stryder.respawn.com",
 						port: 443,
 						path: `/nucleus-oauth.php?qt=origin-requesttoken&type=server_token&code=${ request.query.token }&forceTrial=0&proto=0&json=1&&env=production&userId=${ parseInt( request.query.id ).toString( 16 ).toUpperCase() }`
-					} )
+					} ).catch( reason => authResponse = JSON.stringify( reason ) ) // convert this to a string because the rest of the code expects authResponse to be a string
```

Comment:
This code smells, plus it doesn't work for the reason you think it does: authResponse gets set to the stringified error not because you do `authResponse = ` but rather because that returns the value (it's c-like), which catch converts into the new promise, which fulfills the await, which goes into authResponse. Moreover, since you now catch the error from the only thing in that try block, the catch block after is now useless (it would have been executed for the rejected promise had you not done that thingy with `.catch()`.